packages/block-editor/src/components/spacing-sizes-control/spacing-input-control.js

```diff
@@ -0,0 +1,240 @@
+
+	const getNewCustomValue = ( newSize ) => {
+		const isNumeric = ! isNaN( parseFloat( newSize ) );
+		const nextValue = isNumeric ? newSize : undefined;
+		return nextValue;
+	};
+
+	const getNewPresetValue = ( newSize ) => {
+		setValueNow( newSize );
+		const size = parseInt( newSize, 10 );
+		if ( size === 0 ) {
+			return '0';
+		}
+		return `var:preset|spacing|${ spacingSizes[ newSize ]?.slug }`;
+	};
+
+	const handleCustomValueSliderChange = ( next ) => {
+		onChange(
+			next !== undefined ? `${ next }${ selectedUnit }` : undefined
+		);
+	};
+
+	const allPlaceholder = isMixed ? __( 'Mixed' ) : null;
+
+	const currentValueHint = ! isMixed
+		? customTooltipContent( currentValue )
+		: __( 'Mixed' );
+
+	const options = spacingSizes.map( ( size, index ) => ( {
+		key: index,
+		name: size.name,
+	} ) );
+
+	const marks = spacingSizes.map( ( newValue, index ) => ( {
+		value: index,
+		label: undefined,
+	} ) );
+
+	const ariaLabel = sprintf(
+		// translators: 1: The side of the block being modified (top, bottom, left, etc.). 2. Type of spacing being modified (Padding, margin, etc)
+		__( '%1$s %2$s' ),
+		LABELS[ side ],
+		type?.toLowerCase()
+	);
+
+	return (
+		<>
+			{ side !== 'all' && (
+				<HStack className="components-spacing-sizes-control__side-labels">
+					<Text className="components-spacing-sizes-control__side-label">
+						{ LABELS[ side ] }
+					</Text>
+
+					{ spacingSizes.length <= 9 &&
+						! showCustomValueControl &&
+						currentValueHint !== undefined && (
+							<Text className="components-spacing-sizes-control__hint-single">
+								{ currentValueHint }
+							</Text>
+						) }
+				</HStack>
+			) }
+			{ side === 'all' &&
+				spacingSizes.length <= 9 &&
+				! showCustomValueControl &&
+				currentValueHint !== undefined && (
+					<Text className="components-spacing-sizes-control__hint-all">
+						{ currentValueHint }
+					</Text>
+				) }
+
+			<Button
+				label={
+					showCustomValueControl
+						? __( 'Use size preset' )
+						: __( 'Set custom size' )
+				}
+				icon={ settings }
+				onClick={ () => {
+					setShowCustomValueControl( ! showCustomValueControl );
+				} }
+				isPressed={ showCustomValueControl }
+				isSmall
+				className={ classnames( {
+					'components-spacing-sizes-control__custom-toggle-all':
+						side === 'all',
+					'components-spacing-sizes-control__custom-toggle-single':
+						side !== 'all',
+				} ) }
+				iconSize={ 24 }
+			/>
+			{ showCustomValueControl && (
+				<>
+					<UnitControl
+						onChange={ ( newSize ) =>
+							onChange( getNewCustomValue( newSize ) )
+						}
+						value={ currentValue }
+						units={ units }
+						placeholder={ allPlaceholder }
+						disableUnits={ isMixed }
+						label={ ariaLabel }
+						hideLabelFromVision={ true }
+						className="components-spacing-sizes-control__custom-value-input"
+					/>
+
+					<RangeControl
+						value={ customRangeValue }
+						min={ 0 }
+						max={ 100 }
```

Comment:
Because it is a short range control upping it to something like 200 seems to make it a bit fiddler to get smaller values at the bottom end for the likes of `rem` and `em` values.  I would be inclined to leave it at 100 for now and see what feedback we get.

Reply:
Great point, yes, let's leave it at `100` then.

Reply:
> Because it is a short range control upping it to something like 200 seems to make it a bit fiddler to get smaller values at the bottom end for the likes of rem and em values.

I'd suggest then we simply tweak the `max` prop based on the selected unit. Similar to what is done to adjust the step value for UnitControls based on the current unit. Speaking of `step` should we add that here?